File: src/main/java/phoupraw/mcmod/torches_in_water/config/ItemRendererInConfig.java
package phoupraw.mcmod.torches_in_water.config;

import dev.isxander.yacl3.gui.image.ImageRenderer;
import net.minecraft.client.gui.DrawContext;
import net.minecraft.item.ItemStack;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;
import java.util.function.Supplier;

public record ItemRendererInConfig(Supplier<ItemStack> itemStack) implements RendererInConfig {
    public static CompletableFuture<Optional<ImageRenderer>> of(Supplier<ItemStack> itemStack) {
        return CompletableFuture.completedFuture(Optional.of(new ItemRendererInConfig(itemStack)));
    }
    @Override
    public int render(DrawContext graphics, int x, int y, int renderWidth, float tickDelta) {
        graphics.drawItem(itemStack().get(), x, y);
        return 16;
    }
}

File: src/main/kotlin/phoupraw/mcmod/torches_in_water/config/ItemRendererInConfig.kt
package phoupraw.mcmod.torches_in_water.config

import net.minecraft.client.gui.DrawContext
import net.minecraft.item.ItemStack

fun interface ItemRendererInConfig : RendererInConfig {
    fun itemStack(): ItemStack
    override fun render(graphics: DrawContext, x: Int, y: Int, renderWidth: Int, tickDelta: Float): Int {
        graphics.drawItem(itemStack(), x, y)
        return 16
    }
}